### PR TITLE
add tip generator

### PR DESCRIPTION
**Pull request by 7sne**

### CODE REVIEW

**Review by hasparus — CHANGES_REQUESTED** [review]
🙈

**Review by LogvinovLeon — COMMENTED** [review]
I like the idea in general, before merging - let's add actual tips and add to other pages

File: src/components/Tips.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { useState } from 'react';
+
+import BulbSvg from '../../public/static/svg/bulb';
+
+type TipCategory = 'general' | 'calculator' | 'decoder';
+
+const tips: { [key in TipCategory]: string[] } = {
```
Comment (hasparus) [inline src/components/Tips.tsx:7]: I'd argue this dict is redundant.

Imagine we have a thousand pages.
We'd have to keep all tips here in a huge dictionary, which gets bundled with every other page.
To update a tip, I have to open a page, copy it's ID and then go to this file to look up the ID.

Wouldn't it be easier to keep the calculator tips in the calculator page and the decoder tips in the decoder page?

---

A good rule of thumb for an answer to question "Should I keep it in XML or should I keep it in JSON?" is "Are you building an user interface, or is it serialized data to store or send over the wire?"

**Review by hasparus — APPROVED** [review]

**Comment by hasparus** [comment]
Merging to avoid more conflicts, we can add more tips later.

**Pull request merged**